Core/include/Acts/Vertexing/HelicalTrackLinearizer.ipp

```diff
@@ -61,132 +66,137 @@ Acts::Result<Acts::LinearizedTrack> Acts::
 
   // Extracting Perigee parameters and compute functions of them for later
   // usage
-  double phi = paramsAtPCA(BoundIndices::eBoundPhi);
-  double sinPhi = std::sin(phi);
-  double cosPhi = std::cos(phi);
+  ActsScalar d0 = paramsAtPCA(BoundIndices::eBoundLoc0);
 
-  double theta = paramsAtPCA(BoundIndices::eBoundTheta);
-  const double sinTheta = std::sin(theta);
-  const double tanTheta = std::tan(theta);
+  ActsScalar phi = paramsAtPCA(BoundIndices::eBoundPhi);
+  ActsScalar sinPhi = std::sin(phi);
+  ActsScalar cosPhi = std::cos(phi);
+
+  ActsScalar theta = paramsAtPCA(BoundIndices::eBoundTheta);
+  ActsScalar sinTheta = std::sin(theta);
+  ActsScalar tanTheta = std::tan(theta);
 
   // q over p
-  double qOvP = paramsAtPCA(BoundIndices::eBoundQOverP);
+  ActsScalar qOvP = paramsAtPCA(BoundIndices::eBoundQOverP);
 
-  // Charge of the particle, determines the sign of the helix radius rho
-  double sgnH = (qOvP < 0.) ? -1 : 1;
+  // Get mass hypothesis from propagator options
+  ActsScalar m0 = pOptions.mass;
+  // Assume unit charge
+  ActsScalar p = std::abs(1. / qOvP);
```

Comment:
the charge hypothesis could be acquired by the stepper config which should be accessible from the propagator